File: sdk/js-sdk/src/api/battle.ts
import { AxiosInstance } from "axios";

export class BattleAPI {

    constructor(
        private client: AxiosInstance,
    ) {}

    async list() {

        return (

            await this.client.get("/battle")

        ).data;

    }

    async get(id: string) {

        return (

            await this.client.get(`/battle/${id}`)

        ).data;

    }

    async create(payload: any) {

        return (

            await this.client.post(

                "/battle",

                payload,

            )

        ).data;

    }

}

File: sdk/skillbattle/api/battle.py
class BattleAPI:

    def __init__(

        self,

        client,

    ):

        self.client = client

    def list(self):

        return self.client.get(

            "/battle",

        ).json()

    def get(

        self,

        battle_id,

    ):

        return self.client.get(

            f"/battle/{battle_id}",

        ).json()

    def create(

        self,

        payload,

    ):

        return self.client.post(

            "/battle",

            json=payload,

        ).json()